src/test/java/uk/gov/hmcts/reform/managecase/service/noc/RequestNoticeOfChangeServiceTest.java

```diff
@@ -0,0 +1,319 @@
+        updateCaseResourceData(caseResource, CHANGE_ORGANISATION_REQUEST_KEY, changeOrganisationRequest);
+
+        List<OrganisationPolicy> organisationPolicies = updateCaseResourceWithOrganisationPolicies(caseResource);
+
+        Organisation invokersOrganisation =
+            Organisation.builder().organisationID(INVOKERS_ORGANISATION_IDENTIFIER).build();
+        OrganisationPolicy invokersOrganisationPolicy = new OrganisationPolicy(invokersOrganisation,
+                                                                               ORG_POLICY_REFERENCE,
+                                                                               CASE_ASSIGNED_ROLE);
+
+        updateCaseResourceData(caseResource, ORGANISATION_POLICY_KEY, invokersOrganisationPolicy);
+
+        setInvokerToActAsAnAdminOrSolicitor(isAdminOrSolicitor);
+
+        given(jacksonUtils.convertValue(any(JsonNode.class), any()))
+            .willReturn(changeOrganisationRequest)
+            .willReturn(organisationPolicies.get(0))
+            .willReturn(organisationPolicies.get(1))
+            .willReturn(organisationPolicies.get(2))
+            .willReturn(invokersOrganisationPolicy);
+
+        given(dataStoreRepository.findCaseByCaseIdExternalApi(CASE_ID)).willReturn(caseResource);
+    }
+
+    @Test
+    @DisplayName("Generate a Notice Of Change Request with approval and Auto assignment of case roles")
+    void testApprovalComplete() {
+        caseResource.setJurisdiction(JURISDICTION_ONE);
+        nocAutoApprovedByAdminOrSolicitor(true);
+
+        RequestNoticeOfChangeResponse requestNoticeOfChangeResponse
+            = service.requestNoticeOfChange(noCRequestDetails);
+
+        @SuppressWarnings("unchecked")
+        ArgumentCaptor<List<String>> caseRolesCaptor = ArgumentCaptor.forClass(List.class);
+        ArgumentCaptor<String> caseIdCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> userIdCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> organisationIdCaptor = ArgumentCaptor.forClass(String.class);
+
+        verify(dataStoreRepository).assignCase(caseRolesCaptor.capture(),
+                                               caseIdCaptor.capture(),
+                                               userIdCaptor.capture(),
+                                               organisationIdCaptor.capture());
+
+        assertThat(caseRolesCaptor.getValue()).isNotEmpty();
+        assertThat(caseIdCaptor.getValue()).isEqualTo(CASE_ID);
+        assertThat(userIdCaptor.getValue()).isEqualTo(USER_INFO_UID);
+        assertThat(organisationIdCaptor.getValue()).isEqualTo(INVOKERS_ORGANISATION_IDENTIFIER);
+        assertThat(requestNoticeOfChangeResponse.getApprovalStatus()).isEqualTo(APPROVED);
+    }
+
+    @Test
+    @DisplayName("Generate a Notice Of Change Request with approval and Auto assignment of case roles")
```

Comment:
Change display name for both this and proceeding test to show difference: i.e. this one has no assignment of case roles.